src/js/pages/services/task-details/TaskDetail.js

```diff
@@ -0,0 +1,238 @@
+      taskDirectoryErrorCount: 0
+    };
+
+    this.store_listeners = [
+      {name: 'state', events: ['success'], listenAlways: false},
+      {name: 'summary', events: ['success'], listenAlways: false},
+      {name: 'taskDirectory', events: ['error', 'success']}
+    ];
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  componentWillMount() {
+    super.componentWillMount(...arguments);
+    this.tabs_tabs = Object.assign({}, SERVICES_TABS);
+    if (this.props.params.nodeID != null) {
+      this.tabs_tabs = Object.assign({}, NODES_TABS);
+    }
+    this.updateCurrentTab();
+  }
+
+  updateCurrentTab() {
+    let routes = this.context.router.getCurrentRoutes();
+    let currentTab = routes[routes.length - 1].name;
+    if (currentTab != null) {
+      this.setState({currentTab});
+    }
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    this.store_removeEventListenerForStoreID('summary', 'success');
+  }
+
+  onStateStoreSuccess() {
+    let task = MesosStateStore.getTaskFromTaskID(this.props.params.taskID);
+    TaskDirectoryStore.getDirectory(task);
+  }
+
+  onTaskDirectoryStoreError() {
+    this.setState({
+      taskDirectoryErrorCount: this.state.taskDirectoryErrorCount + 1
+    });
+  }
+
+  onTaskDirectoryStoreSuccess() {
+    this.setState({
+      directory: TaskDirectoryStore.get('directory'),
+      taskDirectoryErrorCount: 0
+    });
+  }
+
+  hasLoadingError() {
+    return this.state.taskDirectoryErrorCount >= 3;
+  }
+
+  getErrorScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center vertical-center inverse">
+        <RequestErrorMsg />
+      </div>
+    );
+  }
+
+  getLoadingScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center vertical-center inverse">
+        <div className="row">
+          <div className="ball-scale">
+            <div />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  handleOpenLogClick(selectedLogFile) {
+    this.setState({selectedLogFile, currentTab: 'debug'});
+  }
+
+  getBasicInfo() {
+    let task = MesosStateStore.getTaskFromTaskID(this.props.params.taskID);
+    if (task == null) {
+      return null;
+    }
+
+    let taskIcon = (
+      <img src={task.getImages()['icon-large']} />
+    );
+
+    let tabs = (
+      <ul className="tabs list-inline flush-bottom container-pod container-pod-short-top inverse">
+        {this.tabs_getRoutedTabs({params: this.props.params})}
+      </ul>
+    );
+
+    return (
+      <PageHeader
+        icon={taskIcon}
+        subTitle={TaskStates[task.state].displayName}
+        navigationTabs={tabs}
+        title={task.getId()} />
+    );
+  }
+
+  tabs_handleTabClick() {
+    this.setState({selectedLogFile: null});
+
+    // Only call super after we are done removing/adding listeners
+    super.tabs_handleTabClick(...arguments);
```

Comment:
Because we are implementing the function our selves to set `selectedLogFile` on tab change, so in order to keep original functionality we call `super` afterwards